### DISCUSSION TITLE
Support Deno

### DISCUSSION THREAD

**Discussion by zadeviggers** [withastro/roadmap#97] [Proposal]
It would be cool if Astro supported [Deno](https://deno.land/).
Supporting Deno would make Astro accessible to a wider audience (and also make Astro seem even cooler!).

If we support Deno, then when server-side rendering comes out it would help enable hosting on [Deno deploy](https://deno.com/deploy).

**Comment by okikio** [comment]
This is something we could do in theory, but it will require a lot more work before it's ready. We could use `browser-vite` together with Deno [`service-workers`](https://deno.land/manual/runtime/workers), but before all that happens we need to try testing `browser-vite` with Astro, which is currently a WIP.

**Comment by FredKSchott** [comment]
cc @matthewp -- I believe Matthew wants Deno + Deno Deploy to be a supported target for our SSR support!

As for the tool itself, we are tied to Vite at the moment, and I believe Vite has a thread where they are not currently interested in supporting Deno. This would make it very hard (impossible) for us to support Deno without them. If they ever do consider revisiting, we should too!

**Reply by matthewp** [reply]
I think this RFC wants Deno as the dev environment, which would indeed be very hard if not impossible given our coupling to Vite. If Vite were to work in Deno, then maybe.

**Reply by zadeviggers** [reply]
After poking through some old Vite issues, I found [this](https://github.com/mathe42/vite-plugin-server) which claims to allow the vite server to run on Deno, although it seems quite hacky.

**Reply by mxdvl** [reply]
Worth noting that Deno Deploy is supported with [@astro/deno](https://github.com/withastro/astro/tree/main/packages/integrations/deno) already. 

For previewing locally through Deno, I would recommend [Elian Van Custem’s tutorial on their blog](https://www.elian.codes/blog/22-05-06-deploying-astro-on-deno/), published last week.

**Comment by ben-laird** [comment]
I'd really like to revisit this in an upcoming Astro release. I came from [this issue](https://github.com/withastro/starlight/issues/1387) where I encountered Astro throwing errors about reading `import.meta.env`, and I thought it was a genuine bug, not a lack of a feature. It's a bit surprising to me since Vite is ok running in a Deno environment in development (ie running `deno task dev`), so I figured Astro would follow suit.

It just seems really odd to support Deno in production through [@astrojs/deno](https://github.com/withastro/astro/tree/main/packages/integrations/deno) but not in development. I don't think an adapter should be required either, Astro should run in static mode in a Deno env in dev without throwing any errors, especially ones about a Vite feature.
